### DISCUSSION TITLE
Welcome to cFS Discussions!

### DISCUSSION THREAD

**Discussion by JustinGOSSES** [nasa/cFS#269] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by skliper** [comment]
Hi all!  I'm Jake and my current role is the product development lead for the core Flight System framework.  My goal is to maintain and improve the framework for all stakeholders and the community, foster innovation, and promote collaboration.

**Comment by quinnha** [comment]
Hey all! I'm Quinn, a Canadian software engineering student at McMaster University in Hamilton, Ontario. I'm currently working on a FreeRTOS port for the OSAL, which is proving to be a lot more of a task than I originally thought 😆. 
Feel free to connect with me here: https://www.linkedin.com/in/quinn-ha/

**Comment by tstern-masten** [comment]
Hello everyone! My name is Tim and I am a recently hired flight software and platforms engineer with Masten Space Systems! This is my first venture into flight software and I am looking forward to learning more and engaging with the community!

**Reply by tstern-masten** [reply]
Also I am wondering if anyone has used/is using the https://github.com/nasa/CF app for CFDP? If not are there other alternatives people have used?

**Reply by astrogeco** [reply]
Welcome! A new version of CF is coming soon!

**Reply by skliper** [reply]
CF has been used on previous missions, but the open source version currently available is a bit challenging to work with.  I think @dmccomas has it working in https://github.com/OpenSatKit/OpenSatKit as a sample setup.  As @astrogeco mentioned, a new version is coming and it's targeted for use on a few missions currently in development.

**Reply by tstern-masten** [reply]
Awesome! I will keep my eyes peeled for that update. Thank you!

**Reply by dmccomas** [reply]
Welcome to the community!  

The default file transport in OpenSatKit is TFTP but CFDP is available. The CFDP ground engine is always started and the CF app is loaded when you start the cFS from the "Mission FSW" tab.

You can run a demo that uses CFDP to transfer files from flight to ground.  To access the demo
1. Select OSK's main screen Mission FSW tab and start the cFS. You must use the "Mission FSW" cFS target
2. On the Mission FSW screen select "Data/File Mgmt" button in the Application Groups
3. From the Data/File Management screen' Learning Resources  section choose "Data-File Mgmt Script" from the "Run Demo" drop down menu and then select "Run Demo"
- This runs a script that configures apps to generate files and then transfer the files from flight to ground using either TFTP or CFDP, your choice.

Even if you don't install OSK and run the demo the scripts may be helpful. COSMOS uses Ruby but they're pretty easy to follow even if you don't know Ruby. The main demo is script is at

https://github.com/OpenSatKit/OpenSatKit/blob/master/cosmos/config/targets/SIMSAT/procedures/demo_datafile_mgmt.rb

The recorder playback function is in the following script. 

https://github.com/OpenSatKit/OpenSatKit/blob/master/cosmos/config/targets/SIMSAT/procedures/simsat_recorder_mgmt.rb

**Reply by tstern-masten** [reply]
Thank you! I will take a look at that for sure!

**Comment by janup2442** [comment]
Hello Everyone , I am Anup Jaiswal , an Undergraduate engineering student at **Madan Mohan Malaviya University Of Technology (MMMUT), Gorakhpur , India** . I am intrested in Robotics , Software Engineering , eSports and AI/ML . Recently , we decided to work on a CubeSat project , currently we are doing research on it . As I am new in spacecraft technology and want to learn , develop and contribute to this field . My role in this project is to develop OBC and Flight Software for the cubesat . 
My specialisation is Electrical and Comouter science engineering. 

I am struggling with tools and available resources as there are lot of options for software and hardware . I need help . 
If someone is intrested to discuss / collaborate / develop along with me . It will be helpful fot both of us and this community as well. 

I am llooking forware to connect with Intrested university /college students or industry pofessionals on this mission . 
Thank you .

**Comment by rishabhsingroha** [comment]
hello everyone I am Rishabh a fullstack + AIML dev 
been working on a lot of SAAS 
contect if you want to be a part
